src/ray/raylet/worker_pool.cc

```diff
@@ -118,12 +118,18 @@ void WorkerPool::StartWorkerProcess(const Language &language) {
 
   // Launch the process to create the worker.
   pid_t pid = fork();
-  if (pid != 0) {
+  if (pid < 0) {
+    // Failure case.
+    RAY_LOG(FATAL) << "Failed to create new process: " << strerror(errno);
```

Comment:
better be more specific, `Failed to fork worker process: `

Reply:
This is related to https://github.com/ray-project/ray/issues/476 (the issue is about the local scheduler, but I think the raylet also has this issue).

Do you see a change in behavior if you create too many workers really quickly? I would expect:
- before this PR, the raylet fails to create some workers, but the raylet does not die
- after this PR, the raylet will die in that situation